File: other/MT1051/week03/HW07/HW07_109601003.py
# -*- coding: utf-8 -*-
"""
Date: 2023/09/25
HW: 07
Author: 林群賀
Student Number: 109601003
"""

import math
from typing import Any

class ECalculator:

    def __call__(self, *args: Any, **kwds: Any) -> Any:
        pass

    def __init__(self):
        pass

    def calculate(self, num):
        if num < 50:
            num = math.sqrt(num) * 10
        else:
            num += 12
        return num

def main() -> None:
    e_calculator = ECalculator()
    num = float(input("Please input the value E: "))
    result = e_calculator.calculate(num)
    print("The final result of E:", result)

if __name__ == '__main__':
    main()

File: mt/mt1051/week03/HW07/HW07_109601003.go
// Date: 2023/09/25
// HW: 07
// Author: 林群賀
// Student Number: 109601003

package main

import (
	"fmt"
	"math"
)

func main() {
	// 输入一个数值 E
	var E float64
	fmt.Print("请输入一个数值 E: ")
	fmt.Scan(&E)

	// 根据条件进行操作
	var result float64
	if E < 50 {
		result = math.Sqrt(E) * 10
	} else {
		result = E + 12
	}

	// 输出最终结果
	fmt.Println("最终结果为:", result)
}
